Deprecate Flow/Stream Java API (PR 1/4)

sreevatsanraman opened the pull request:
PR 1/4
- [x] Deprecate Stream and Flow in Java APIs

Future PRs 
- [ ] Deprecate CLI
- [ ] Modify Examples
- [ ] Change Docs

Commit:
Deprecate Flow/Stream Java API

Commit:
Change stream to realtime

poornachandra commented on cdap-api/src/main/java/co/cask/cdap/api/DatasetConfigurer.java:
```diff
@@ -32,13 +32,15 @@
    *
    * @param stream {@link Stream}
    */
+  @Deprecated
```
It would be good to add a deprecation javadoc like the follows (over here and at all other places below) - 
```
/**
 * @deprecated  As of release 5.0, replaced by {@link replacement-link}
 */
```

sreevatsanraman replied:
Addressed comments

poornachandra commented on cdap-api/src/main/java/co/cask/cdap/api/DatasetConfigurer.java:
```diff
@@ -32,13 +32,15 @@
    *
    * @param stream {@link Stream}
    */
+  @Deprecated
```
I notice that many places below have the deprecated javadoc 👍

sreevatsanraman replied:
Fixed this instance.

poornachandra commented on cdap-api/src/main/java/co/cask/cdap/api/annotation/Batch.java:
```diff
@@ -99,9 +99,12 @@
  *
  * @see HashPartition
+ *
+ * @deprecated As of release 5.0.0, implement {@link @ProgramType.SPARK} for realtime processing.
```
Instead of implement, maybe say "replaced by"?

sreevatsanraman replied:
Addressed comments

poornachandra reviewed (commented):
@sreevatsanraman Overall the changes look good. Just a couple of comments

Commit:
Address comments

sreevatsanraman commented:
Addressed comments. Compile build passes locally

```
INFO] CDAP Spark Scala Archetype ......................... SUCCESS [  0.289 s]
[INFO] CDAP MapReduce Archetype ........................... SUCCESS [  0.165 s]
[INFO] CDAP Application Archetype ......................... SUCCESS [  0.161 s]
[INFO] CDAP Integration Test Framework .................... SUCCESS [  0.683 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 02:56 min
[INFO] Finished at: 2018-07-10T08:49:21-07:00
[INFO] Final Memory: 350M/1432M
[INFO] ------------------------------------------------------------------------
```

poornachandra commented on cdap-api/src/main/java/co/cask/cdap/api/DatasetConfigurer.java:
```diff
@@ -32,13 +32,16 @@
    * @param stream {@link Stream}
    */
+  @Deprecated
   void addStream(Stream stream);
```
Missing `@deprecated` javadoc for this method

sreevatsanraman replied:
Added.

poornachandra reviewed (commented):
@sreevatsanraman  Just one minor comment, please fix and merge

sreevatsanraman commented:
Fixed the comments. Local compile build passes. Merging.
